File: app/models/vault.py
from datetime import datetime

from sqlalchemy import Column, String, DateTime, JSON, Boolean, CHAR

from ..database import Base


class Vault(Base):
    __tablename__ = "vaults"
    id = Column(CHAR(36), primary_key=True, index=True)
    name = Column(String(255), index=True)
    description = Column(String(255))
    user_id = Column(CHAR(36))
    created_at = Column(DateTime, default=datetime.utcnow, nullable=False)
    tags = Column(JSON, nullable=True)
    is_default = Column(Boolean, default=False)

File: go_app/server/schemas/vault.go
package schemas

import "time"

type Vault struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Description string    `default:"" json:"description"`
	UserID      string    `json:"userId"`
	CreatedAt   time.Time `json:"createdAt"`
	Tags        []string  `json:"tags"`
	IsDefault   bool      `default:"false" json:"isDefault"`
}
